xrft/xrft.py

```
@@ -641,30 +663,65 @@ def power_spectrum(da, dim=None, scaling="density", **kwargs):
         {"true_amplitude": True, "true_phase": False}
     )  # true_phase do not matter in power_spectrum
 
-    daft = dft(da, dim=dim, **kwargs)
+    daft = dft(da, dim=dim, real=real, **kwargs)
     updated_dims = [
         d for d in daft.dims if (d not in da.dims and "segment" not in d)
     ]  # Transformed dimensions
     ps = np.abs(daft) ** 2
 
+    if real is not None:
+        real_dim = [d for d in updated_dims if real == d[-len(real) :]][
+            0
+        ]  # find transformed real dimension
+        f = np.full(ps.sizes[real_dim], 2.0)
+        if len(da[real]) % 2 == 0:
+            f[0], f[-1] = 1.0, 1.0
+        else:
+            f[0] = 1.0
+        ps = ps * xr.DataArray(f, dims=real_dim)
+
     if scaling == "density":
+        if window_correction:
+            for arg in kwargs.keys():
+                if arg == "window":
+                    if kwargs[arg] is None:
+                        raise ValueError("Windowing needs to be turned on.")
```

Maybe a little more verbose, e.g
"window_correction can only be applied when a non-uniform window is applied"
or, this could just silently do nothing?

---

Silently doing nothing is definitely going to confuse users.